### PR TITLE
Refine streaming support

### PR DESCRIPTION
**Pull request by KaQuMiQ**

### CODE REVIEW

File: src/draive/openai/chat.py [hunk 1]
```diff
@@ -46,34 +61,58 @@ async def openai_chat_completion(
...
+                        instruction=instruction,
+                        history=history or [],
+                        input=input,
+                    ),
+                    toolset=toolset,
+                )
+
+            case True:
+
+                async def stream_task(
```
Comment (piotr-rarus) [inline src/draive/openai/chat.py:84]: I'd make this one a private method, same as `_chat_stream`. Somehow I always dislike defining methods inside methods :/

File: src/draive/openai/chat_response.py [hunk 2]
```diff
@@ -22,11 +22,11 @@
 from draive.types import Toolset
 
 __all__ = [
-    "_chat_completion",
+    "_chat_response",
```
Comment (piotr-rarus) [inline src/draive/openai/chat_response.py:25]: Why are we exposing a private method?

File: src/draive/openai/conversation.py [hunk 3]
```diff
@@ -181,21 +76,91 @@ async def _openai_conversation_stream(
         else:
             history = []
 
-        actions: dict[str, ConversationStreamingAction] = {}
+        match stream:
+            case True:
+
+                @ctx.with_current
+                async def stream_task(
```
Comment (piotr-rarus) [inline src/draive/openai/conversation.py:83]: ditto

File: src/draive/openai/chat.py (hunk 1 above)
Comment (KaQuMiQ) [reply src/draive/openai/chat.py:85]: it is more than private 😉  and it captures the context - can't be defined outside

**Pull request merged**